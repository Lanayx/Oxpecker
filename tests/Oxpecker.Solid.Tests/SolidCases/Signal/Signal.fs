module Oxpecker.Solid.Tests.Cases.Signal

open Oxpecker.Solid

[<SolidComponent>]
let Test () =
    let count, setCount = createSignal 0

    Fragment() {
        p() { $"Count1 is {count()}" }
        p() { "Count2 is "; count() }
        button(class'="button", onClick= fun _ -> count() + 1 |> setCount) { "Click me!" }
    }
