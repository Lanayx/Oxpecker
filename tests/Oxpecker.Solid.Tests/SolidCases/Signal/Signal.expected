import { createSignal } from "solid-js";

export function Test() {
    const patternInput = createSignal(0);
    const count = patternInput[0];
    return <>
        <p>
            {`Count1 is ${count()}`}
        </p>
        <p>
            Count2 is 
            {count()}
        </p>
        <button class="button"
            onClick={(_arg) => {
                patternInput[1](count() + 1);
            }}>
            Click me!
        </button>
    </>;
}

